File: app/src/main/java/net/xzos/UpgradeAll/ui/viewmodels/log/SectionsPagerAdapter.java
package net.xzos.UpgradeAll.ui.viewmodels.log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.LiveData;

import net.xzos.UpgradeAll.server.ServerContainer;
import net.xzos.UpgradeAll.server.log.LogDataProxy;

import java.util.ArrayList;
import java.util.List;

/**
 * A [FragmentPagerAdapter] that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
public class SectionsPagerAdapter extends FragmentStatePagerAdapter {

    private static ArrayList<String> TAB_TITLES = new ArrayList<>();
    private String logSort;
    private List<String> mLogObjectIdList;

    public SectionsPagerAdapter(LifecycleOwner owner, FragmentManager fm, String logSort) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        this.logSort = logSort;
        LogDataProxy logDataProxy = new LogDataProxy(ServerContainer.AppServer.getLog());
        LiveData<List<String>> liveDataLogObjectIdList = logDataProxy.getLiveDataLogObjectIdList(logSort);
        liveDataLogObjectIdList.observe(owner, logObjectIdList -> {
            TAB_TITLES.clear();
            for (String databaseIdString : logObjectIdList) {
                String name = LogDataProxy.getNameFromId(databaseIdString);
                TAB_TITLES.add(name);
            }
            mLogObjectIdList = logObjectIdList;
            notifyDataSetChanged();
        });
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        String[] logObjectTag = {logSort, mLogObjectIdList.get(position)};
        return PlaceholderFragment.newInstance(logObjectTag);
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return TAB_TITLES.get(position);
    }

    @Override
    public int getItemPosition(@NonNull Object object) {
        return POSITION_NONE;
    }

    @Override
    public int getCount() {
        return TAB_TITLES.size();
    }
}

File: app/src/main/kotlin/net/xzos/upgradeAll/ui/viewmodels/log/SectionsPagerAdapter.kt
package net.xzos.upgradeAll.ui.viewmodels.log

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.Observer
import androidx.viewpager.widget.PagerAdapter
import net.xzos.upgradeAll.server.ServerContainer
import net.xzos.upgradeAll.server.log.LogDataProxy
import java.util.*

class SectionsPagerAdapter(owner: LifecycleOwner, fm: FragmentManager, private val logSort: String) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    private var mLogObjectIdList: List<String>? = null

    init {
        val logDataProxy = LogDataProxy(ServerContainer.AppServer.log)
        val liveDataLogObjectIdList = logDataProxy.getLiveDataLogObjectIdList(logSort)
        liveDataLogObjectIdList.observe(owner, Observer { logObjectIdList ->
            TAB_TITLES.clear()
            for (databaseIdString in logObjectIdList) {
                val name = LogDataProxy.getNameFromId(databaseIdString)
                if (name != null) TAB_TITLES.add(name)
            }
            mLogObjectIdList = logObjectIdList
            notifyDataSetChanged()
        })
    }

    override fun getItem(position: Int): Fragment {
        val logObjectTag = arrayOf(logSort, mLogObjectIdList!![position])
        return PlaceholderFragment.newInstance(logObjectTag)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return TAB_TITLES[position]
    }

    override fun getItemPosition(`object`: Any): Int {
        return PagerAdapter.POSITION_NONE
    }

    override fun getCount(): Int {
        return TAB_TITLES.size
    }

    companion object {

        private val TAB_TITLES = ArrayList<String>()
    }
}